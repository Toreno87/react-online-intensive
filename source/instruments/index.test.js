//Core
import { sum, delay, getUniqueID } from './';

describe('instruments:', () => {
    test('sum function should be a function', () => {
        expect(sum).toBeInstanceOf(Function);
    });

    test('sum function should be thow, when called with not-number type as second argument', () => {
        expect(() => sum(2, 'Привет')).toThrow();
    });

    test('sum function should be thow, when called with not-number type as first argument', () => {
        expect(() => sum('Привет', 2)).toThrow();
    });

    test('sum function should return an additional two arguments passed', () => {
        expect(sum(2, 3)).toBe(5);
        expect(sum(8, 1)).toMatchSnapshot();
    });

    test('delay function should be a returned promise', async () => {
        await expect(delay()).resolves.toBeUndefined();
    });

    test('getUniqueID should be a function', () => {
        expect(getUniqueID).toBeInstanceOf(Function);
    });

    test('getUniqueID function should be thow, when called with not-number type an argument', () => {
        expect(() => getUniqueID('Привет')).toThrow();
    });

    test('getUniqueID function should produce a string of a desired a given length', () => {
        expect(typeof getUniqueID()).toBe('string');
        expect(getUniqueID(5)).toHaveLength(5);
    });
});

