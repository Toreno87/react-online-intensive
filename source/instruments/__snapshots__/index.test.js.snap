// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instruments: sum function should return an additional two arguments passed 1`] = `9`;
